### ISSUE TITLE
Investigate question version update data integrity

### ISSUE THREAD

**Issue by shanemc-goog** [civiform/civiform#2047]
We've had two cases so far where versions_questions hasn't been updated properly.

#1797 and #2040  In both cases other questions used by a program had their versions advanced but not some in both cases.

This causes blocking failures in CiviForm and would seem to indicate a data integrity issue when updating questions related to a program.  It may be that a transaction is not being used as intended or that one needs to.

Note: This has become a general roundup of issues around configuration being updated improperly.

### User Errors to investigate
- [ ] #1797 
- [x] #2040  (#2249)

### Identified Issues
- [x] Editing the same active program in different tabs doesn't work. #2247 
- [ ] Editing the same active question in different tabs doesn't work. #2497
- [x] Deleting an enumerator draft doesn't update other Question drafts that reference it. #2249
- [x] Question Bank doesn't update with new drafts.  Tab 1 Edit program, Tab 2 edit a question in another tab, Tab 1 add question to program, publish, site break because Tab 1 was using the Active not the Draft question. #2375
- [ ] Question [discardDraft has 2 phases ](https://github.com/seattle-uat/civiform/blob/65b86af414afc082037b0b285fd5849e3a019248/universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java#L159)(remove question draft, and update dependencies) and error between these will leave the data corrupted.  The first needs to be committed for the second part to "just work" as it relies on the existing state of the DB.

#### Program Editing specific issues
- [x] #2402

**Comment by gwendolyngoetz** [comment]
#2179 may also be related

**Comment by shanemc-goog** [comment]
I diagnosed #2179 and left details there. the root cause is when an out of date view of a program is used to make a new draft when one had otherwise already been made since the page was loaded.  Unsure the behavior in other inconsistencies like if a draft is already made but has been updated in a different tab etc.

Seemingly questions would have the same problem, but their only interdependency is limited to enumerators.
